tee-worker/identity/litentry/core/omni-account/src/lib.rs

```diff
@@ -0,0 +1,50 @@
+// Copyright 2020-2024 Trust Computing GmbH.
+// This file is part of Litentry.
+//
+// Litentry is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Litentry is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Litentry.  If not, see <https://www.gnu.org/licenses/>.
+
+#![cfg_attr(not(feature = "std"), no_std)]
+
+pub extern crate alloc;
+
+#[cfg(all(not(feature = "std"), feature = "sgx"))]
+extern crate sgx_tstd as std;
+
+#[cfg(all(feature = "std", feature = "sgx"))]
+compile_error!("feature \"std\" and feature \"sgx\" cannot be enabled at the same time");
+
+mod repository;
+pub use repository::*;
+
+mod in_memory_store;
+pub use in_memory_store::InMemoryStore;
+
+use alloc::{collections::btree_map::BTreeMap, vec::Vec};
+use itp_types::parentchain::{AccountId, Hash, Header, ParentchainId};
+use litentry_primitives::MemberAccount;
+
+pub type OmniAccounts = BTreeMap<AccountId, OmniAccountMembers>;
+
+#[derive(Debug, Clone)]
+pub struct OmniAccountMembers {
+	pub member_accounts: Vec<MemberAccount>,
```

Comment:
I think internally stored member_accounts have to be clear text, otherwise you won't be able to compute the hash and pass it as parameter when e.g. `add_account`

Reply:
My understanding is this: The user wants to add an account, so it sends the trusted call to the worker with the `Identity` of the account to be added, with that `Identity` the worker construct the `MemberAccount` to be sent to the parachain (`add_account` extrinsic). The hash is computed from the `Identity` received in the trusted call and the `MemberIdentity` would be `MemberIdentity::Private(identity_encrypted)`. Does it makes sense?